api/redhatopenshift/resource-manager/Microsoft.RedHatOpenshift/preview/2024-06-10-preview/openapi.json

```
@@ -516,19 +519,344 @@
           }
         },
         "x-ms-examples": {
-          "HcpOpenShiftClusters_AdminCredentials_MaximumSet_Gen": {
-            "$ref": "./examples/HcpOpenShiftClusters_AdminCredentials_MaximumSet_Gen.json"
+          "HcpOpenShiftClusters_AdminCredentials_MaximumSet_Gen": {
+            "$ref": "./examples/HcpOpenShiftClusters_AdminCredentials_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters/{hcpOpenShiftClusterName}/kubeConfig": {
+      "post": {
+        "operationId": "HcpOpenShiftClusters_KubeConfig",
+        "tags": [
+          "HcpOpenShiftClusters"
+        ],
+        "description": "Return the kubeconfig for the cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "hcpOpenShiftClusterName",
+            "in": "path",
+            "description": "Name of HCP cluster",
+            "required": true,
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 24,
+            "pattern": "^[a-zA-Z0-9-]{3,24}$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/HcpOpenShiftClusterKubeconfig"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "HcpOpenShiftClusters_KubeConfig_MaximumSet_Gen": {
+            "$ref": "./examples/HcpOpenShiftClusters_KubeConfig_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters/{hcpOpenShiftClusterName}/nodePools": {
+      "get": {
+        "operationId": "NodePools_ListByHcpOpenShiftClusterResource",
+        "tags": [
+          "NodePools"
+        ],
+        "description": "List HcpOpenShiftClusterNodePoolResource resources by HcpOpenShiftClusterResource",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
```

```suggestion
            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
```
I believe if you rebase and then rebuild the latest devcontainer, a fresh `tsp compile` will make this change and more.